src/main/java/seedu/dietmanager/DailyFoodRecord.java

```diff
@@ -100,16 +101,114 @@ public String showLunch() {
+     * @return ArrayList of all the foods consumed in a day
+     */
+    public ArrayList<Food> getDailyFood(String timeFrame) {
+        switch (timeFrame) {
+        case "morning":
+            return breakfast;
+        case "afternoon":
+            return lunch;
+        case "night":
+            return dinner;
+        default:
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Calculate the total recorded calories for all the meals.
+     * @return Optional double of the calories intake if exist, and Optional empty otherwise.
+     */
+    public Optional<Double> getDailyCalories() {
+        ArrayList<Food> allDailyFood = getDailyFood();
+        return allDailyFood.stream().filter(Food::hasCaloriesData)
+                .map(Food::getCalories)
+                .map(Optional::get)
+                .reduce(Double::sum);
+    }
+
+    /**
+     * Calculate the total recorded calories for meals in a specific time frame of a day .
+     * @return Optional double of the calories intake if exist, and Optional empty otherwise.
+     */
+    public Optional<Double> getDailyCalories(String timeFrame) {
+        ArrayList<Food> foodInTimeFrame = getDailyFood(timeFrame);
+        return foodInTimeFrame.stream().filter(Food::hasCaloriesData)
+                .map(Food::getCalories)
+                .map(Optional::get)
+                .reduce(Double::sum);
+    }
+
+    /**
+     * Check if the DailyFoodRecord has any Food with missing calories data.
+     * @return False if there exist food without calories, true otherwise.
+     */
+    public boolean isCaloriesPresent() {
+        ArrayList<Food> allDailyFood = getDailyFood();
+        return allDailyFood.stream().filter(food -> !food.hasCaloriesData())
+                .map(food -> false)
+                .reduce((first, second) -> first)
+                .orElse(true);
+    }
+
+    /**
+     * Check if the DailyFoodRecord has any Food in a specific time frame of a day with missing calories data.
+     * @return False if there exist food without calories, true otherwise.
+     */
+    public boolean isCaloriesPresent(String timeFrame) {
+        ArrayList<Food> allDailyFood = getDailyFood(timeFrame);
+        return allDailyFood.stream().filter(food -> !food.hasCaloriesData())
+                .map(food -> false)
+                .reduce((first, second) -> first)
+                .orElse(true);
+    }
+
+    /**
+     * Express the calculable calories intake for all the meals as a String.
+     * @return String representation of the calculable calories.
+     */
+    public String showDailyCalories() {
+        return getDailyCalories()
+                .map(calories -> MessageBank.CALORIES_MESSAGE + String.format("%.2f", calories) + "cal.\n")
+                .map(result -> (isCaloriesPresent()) ? result : (result + MessageBank.MISSING_CALORIES_MESSAGE))
+                .orElse(MessageBank.NO_CALORIES_MESSAGE);
+    }
```

Comment:
can streamline further to make it more intuitive

Reply:
i try, but this is really quite streamline alr, i see what i can change